tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/contact/EmailAddressContact.scala

```diff
@@ -1,24 +1,29 @@
 package com.linagora.tmail.james.jmap.contact
 
-import java.nio.charset.StandardCharsets
-import java.util.UUID
-
 import com.google.common.collect.{HashMultimap, Multimap, Multimaps}
-import org.apache.james.core.MailAddress
+import org.apache.james.core.{Domain, MailAddress, Username}
 import org.apache.james.jmap.api.model.AccountId
 import org.reactivestreams.Publisher
 import reactor.core.scala.publisher.{SFlux, SMono}
 
+import java.nio.charset.StandardCharsets
+import java.util.UUID
 import scala.jdk.CollectionConverters.CollectionHasAsScala
 
 object EmailAddressContact {
   private def computeId(mailAddress: MailAddress): UUID = UUID.nameUUIDFromBytes(mailAddress.asString().getBytes(StandardCharsets.UTF_8))
 
   def of(address: MailAddress): EmailAddressContact = EmailAddressContact(computeId(address), address)
+
+  def of(address: MailAddress, firstname: String, surname: String): EmailAddressContact =
+    EmailAddressContact(computeId(address), address, firstname, surname)
 }
 
-case class EmailAddressContact(id: UUID, address: MailAddress) {
-  def contains(part: String): Boolean = address.asString().contains(part)
+case class EmailAddressContact(id: UUID, address: MailAddress, firstname: String = "", surname: String = "") {
+  def contains(part: String): Boolean =
+    address.asString().contains(part) ||
+      (firstname != null && firstname.contains(part)) ||
+      (surname != null && surname.contains(part))
```

Comment:
Maybe there is better?

Reply:
IMO we should get rid of the null checks.